tiny_cnn/layers/average_pooling_layer.h

```diff
@@ -147,12 +226,12 @@ class average_pooling_layer : public partial_connected_layer<Activation> {
         cnn_size_t dxmax = std::min(pooling_size, in_.width_ - x);
         cnn_size_t dstx = x / stride_;
         cnn_size_t dsty = y / stride_;
-
+        cnn_size_t outidx = out_.get_index(dstx, dsty, inc);
         for (cnn_size_t dy = 0; dy < dymax; ++dy) {
             for (cnn_size_t dx = 0; dx < dxmax; ++dx) {
                 this->connect_weight(
                     in_.get_index(x + dx, y + dy, inc),
-                    out_.get_index(dstx, dsty, inc),
+                    outidx,
                     inc);
             }
         }
```

Comment:
So should I extract `average_pooling_layer::forward_propagation` and `average_pooling_layer::back_propagation` methods and create below files?

- tiny_average_pooling_kernel.h
- tiny_average_pooling_back_kernel.h

_Optional_
- nnp_average_pooling_kernel.h
- nnp_average_pooling_back_kernel.h
- avx_average_pooling_kernel.h
- avx_average_pooling_back_kernel.h

If certain kernel is not implemented in custom backend, it would be preferrable to fallback into standard kernel (tiny_....)